doc/modules/preprocessing.rst

```diff
@@ -424,5 +425,8 @@ Note that, here, missing values are encoded by 0 and are thus implicitly stored
 in the matrix. This format is thus suitable when there are many more missing
 values than observed values.
 
+Also, knn imputation strategy will use samples with full features, and if all samples
```

Comment:
When was this decided upon? It seems an unnecessary constraint, even if it provides speed.